src/pages/product/ProductPage.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from "react";
+import "./productPage.css";
+import { useNavigate, useParams } from "react-router";
+import { useAuth, useCart, useData, useWishlist } from "../../context";
+
+import {
+  calcPercentage,
+  isProductInCart,
+  isProductInWishlist,
+} from "../../utils";
+
+const ProductPage = () => {
+  const [productQty, setProductQty] = useState(1);
+  const { productId } = useParams();
+  const { productsList } = useData();
+  const { wishlist, removeFromWishlist, addToWishlist, wishlistDispatch } =
+    useWishlist();
+  const { cartList, addToCart, cartDispatch } = useCart();
+
+  const navigate = useNavigate();
+  const { token } = useAuth();
+
+  const product = productsList?.find((product) => {
+    return product._id === productId;
+  });
+
+  const isInCart = isProductInCart(cartList, product?._id);
+  const isInWishlist = isProductInWishlist(wishlist, product?._id);
+
+  const cartHandler = () => {
+    token
+      ? isInCart
+        ? navigate("/cart")
+        : addToCart(cartDispatch, product, token)
+      : navigate("/login");
+  };
+
+  const wishlistHandler = () => {
+    token
+      ? isInWishlist
+        ? removeFromWishlist(product?._id, wishlistDispatch, token)
+        : addToWishlist(wishlistDispatch, product, token)
+      : navigate("/login");
+  };
+  return (
+    <section className="flex-row-center items-start wrap mt-2p5 product-section">
+      <img src={product?.imgURL} />
```

Comment:
suggestion - you may add `alt` too in product.

Reply:
Yess. Thanks for pointing it out. Will update it